### PR TITLE
Improve rates handling

### PR DESCRIPTION
**Pull request by teran-mckinney**
If connecting over Tor, Bitpay would 400 but we would not fallback to Blockchain.info.

### CODE REVIEW

**Comment by teran-mckinney** [comment]
Never used the review request feature before. Hope it's alright with you both.

File: bit/network/rates.py [hunk 1]
```diff
@@ -103,7 +103,10 @@ class BitpayRates:
 
     @classmethod
     def currency_to_satoshi(cls, currency):
-        rate = requests.get(cls.SINGLE_RATE + currency).json()['rate']
+        r = requests.get(cls.SINGLE_RATE + currency)
+        if r.status_code != 200:
```
Comment (ofek) [inline bit/network/rates.py]: `r.raise_for_status()`

File: bit/network/rates.py [hunk 2]
```diff
@@ -196,7 +199,10 @@ class BlockchainRates:
 
     @classmethod
     def currency_to_satoshi(cls, currency):
-        rate = requests.get(cls.SINGLE_RATE.format(currency)).json()
+        r = requests.get(cls.SINGLE_RATE.format(currency))
+        if r.status_code != 200:
```
Comment (ofek) [inline bit/network/rates.py]: `r.raise_for_status()`

**Comment by teran-mckinney** [comment]
Thank you! Switched to `.raise_for_status()`

**Comment by teran-mckinney** [comment]
Would you like me to send a PR bumping the version after this? I could use these changes in production.

**Comment by ofek** [comment]
Sure

**Comment by teran-mckinney** [comment]
@bjarnemagnussen do you have a chance to review this?

**Comment by bjarnemagnussen** [comment]
Sorry for the delay!

Great catch! And for me it looks very good! Go ahead an merge it :)

**Comment by teran-mckinney** [comment]
Thank you!

**Pull request merged**